docs/cloud-api/app-dev-guide.md

```diff
@@ -0,0 +1,278 @@
+# Application developer guide
+
+This guide will provide information on how an application developer can leverage the new Generic Cloud Services APIs, in order to be able to properly use the CloudPublisher/CloudSubscriber API, publish a message, being notified of message delivery and of connection status changes.
+
+The Kura [ExamplePublisher](https://github.com/eclipse/kura/tree/develop/kura/examples/org.eclipse.kura.example.publisher) will be used as a reference.
+
+The application should bind itself to a `CloudPublisher` or `CloudSubscriber` instance, this can be done in different ways, such as using OSGi `ServiceTracker`s or by leveraging the Declarative Service layer.
+
+The recommended way to perform this operation is choosing the latter and allowing the user to customize the service references through component configuration.
+
+If the component metatype and definition are structured as described below, the Kura Web UI will show a dedicated widget in component configuration that helps the user to pick compatible `CloudPublisher` or `CloudSubscriber` instances.
+
+1. **Write component definition**
```

Comment:
Enumeration will not work in this list if you don't indent the rest. Maybe those can be subsections.